docs/designs/supporting_pytorch.md

```diff
@@ -0,0 +1,358 @@
+
+`torch.utils.data.DataLoader` is an iterator that provides all these features.
+
+- Batching the data
+- Shuffling the data
+- Load the data in parallel using multiprocessing workers.
+
+As we set `BATCH_SIZE = 5`, 5 data are exported for learning in a step. The result
+is as follows.
+
+```text
+Epoch:  0 | Step:  0 | batch x:  tensor([5., 6., 2., 1., 8.]) | batch y:  tensor([ 6.,  5.,  9., 10.,  3.])
+Epoch:  0 | Step:  1 | batch x:  tensor([ 7.,  3.,  9.,  4., 10.]) | batch y:  tensor([4., 8., 2., 7., 1.])
+Epoch:  1 | Step:  0 | batch x:  tensor([ 8., 10.,  5.,  7.,  9.]) | batch y:  tensor([3., 1., 6., 4., 2.])
+Epoch:  1 | Step:  1 | batch x:  tensor([2., 1., 4., 6., 3.]) | batch y:  tensor([ 9., 10.,  7.,  5.,  8.])
+Epoch:  2 | Step:  0 | batch x:  tensor([10.,  7.,  3.,  5.,  9.]) | batch y:  tensor([1., 4., 8., 6., 2.])
+Epoch:  2 | Step:  1 | batch x:  tensor([1., 6., 4., 2., 8.]) | batch y:  tensor([10.,  5.,  7.,  9.,  3.])
+```
+
+Another advantage of DataLoader is that the data size can be adjusted automatically
+according to the batch size.
+As we set `BATCH_SIZE = 7`, the result is as follows. The remaining data in this
+epoch will be returned at `step = 1`.
+
+```text
+Epoch:  0 | Step:  0 | batch x:  tensor([1., 7., 2., 8., 9., 6., 3.]) | batch y:  tensor([10.,  4.,  9.,  3.,  2.,  5.,  8.])
+Epoch:  0 | Step:  1 | batch x:  tensor([ 5., 10.,  4.]) | batch y:  tensor([6., 1., 7.])
+Epoch:  1 | Step:  0 | batch x:  tensor([1., 6., 7., 5., 2., 9., 8.]) | batch y:  tensor([10.,  5.,  4.,  6.,  9.,  2.,  3.])
+Epoch:  1 | Step:  1 | batch x:  tensor([10.,  3.,  4.]) | batch y:  tensor([1., 8., 7.])
+Epoch:  2 | Step:  0 | batch x:  tensor([ 4.,  1.,  9.,  5.,  7.,  8., 10.]) | batch y:  tensor([ 7., 10.,  2.,  6.,  4.,  3.,  1.])
+Epoch:  2 | Step:  1 | batch x:  tensor([3., 2., 6.]) | batch y:  tensor([8., 9., 5.])
+```
+
+```python
+import torch
+import torch.utils.data as Data
+
+BATCH_SIZE = 5
+x = torch.linspace(1, 10, 10)       # this is x data (torch tensor)
+y = torch.linspace(10, 1, 10)       # this is y data (torch tensor)
+
+torch_dataset = Data.TensorDataset(x, y)
+loader = Data.DataLoader(
+    dataset=torch_dataset,      # torch TensorDataset format
+    batch_size=BATCH_SIZE,      # mini batch size
+    shuffle=True,               # random shuffle for training
+    num_workers=2,              # subprocesses for loading data
+)
+def show():
+    for epoch in range(3):   # train entire dataset 3 times
+        for step, (batch_x, batch_y) in enumerate(loader):  # for each training step
+            # train your data...
+            print('Epoch: ', epoch, '| Step: ', step, '| batch x: ',
+            batch_x, '| batch y: ', batch_y)
+            # print('Epoch: ', epoch, '| Step: ', step, '| batch x: ',
+            #       batch_x.numpy(), '| batch y: ', batch_y.numpy())
+if __name__ == '__main__':
+    show()
+```
+
+### MNIST Example
+
+This dataset is in numpy array format, and has been stored using pickle, a
```

Comment:
It seems that this documentation is not strongly relaid on the MINIST dataset, can we remove this section?

Reply:
Thank you for your suggestion. I will revise it as soon as possible.